### PR TITLE
Support for multiple files

### PR DESCRIPTION
**Pull request by ImaginaryLost**
Demo of this PR Here: https://ImaginaryLost.github.io/bfv

The main addition of this PR is support to add multiple CSV's at once and be able to switch between them and any sparks with a dropdown button.

Also added is [React Bootstrap](https://www.npmjs.com/package/reactstrap) for use of the button dropdown, [papaparse](https://www.npmjs.com/package/papaparse) for easier parsing of the CSV's, and [lodash](https://www.npmjs.com/package/lodash) for its shuffle function. 

This also switches the Previous/Next button to Left and Right icons from [React Icons](https://www.npmjs.com/package/react-icons)

I understand this PR has a large scope, let me know if there are any parts that need changed/excluded.

### CODE REVIEW

File: src/FileInput.js [hunk 1]
```diff
@@ -1,14 +1,32 @@
...
+        reader.onload = event => {
+            const csvData = PapaParse.parse(
+              event.target.result,
+            );
+            this.props.onFileLoaded(csvData.data, filename);
+          };
+
+        reader.readAsText(value);
+    }
+  }
```
Comment (bf-v) [inline src/FileInput.js]: it looks like papaparse can [parse the csv](https://www.papaparse.com/docs#local-files) without needing a separate FileReader instance, wondering if you gave any thought to this

File: src/FileInput.js [hunk 2]
```diff
@@ -1,14 +1,32 @@
 import React, { Component } from "react";
 import "./FileInput.css";
+const PapaParse = require('papaparse/papaparse.min.js');
```
Comment (bf-v) [inline src/FileInput.js]: preferably use import statements for continuity

File: src/App.js [hunk 3]
```diff
@@ -1,26 +1,48 @@
...
-        .slice(1)
-        .filter(x => !!x);
-      this.setState({ file });
-    });
-    this.state = { file: null };
+    
+    this.state = { 
+      files: null,
+      types: []
+    };
```
Comment (bf-v) [inline src/App.js]: is it necessary to maintain a separate array of types? if i'm not mistaken `types === Object.keys(files)`

File: src/Viewer.js [hunk 4]
```diff
@@ -55,20 +53,46 @@ export default class Viewer extends Component {
...
           </button>
           <button onClick={this.shuffle}>Shuffle</button>
+          <ButtonDropdown isOpen={dropdown} toggle={this.toggle}>
+            <DropdownToggle caret color="black">
+              {currentType}
+            </DropdownToggle>
+            <DropdownMenu>
+              {dropItems}
+            </DropdownMenu>
+          </ButtonDropdown>
```
Comment (bf-v) [inline src/Viewer.js]: The styling for the dropdown looks different than the other buttons, wondering if you caught this:

image

**Comment by ImaginaryLost** [comment]
Closing this PR as its a bit stale, will add another one in a few day when I have time to make changes. New one will deal with merge conflict, PR comments and pull in the mobile changes from https://github.com/bf-v/bfv/pull/4.

**Pull request closed without merging**